langchain/chat/memory.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any, Dict, List, Optional
+
+from langchain.chains.base import Memory
+from langchain.memory.chat_memory import ChatMemory
+
+
+def _get_prompt_input_key(inputs: Dict[str, Any], key: Optional[str]) -> str:
+    if key is not None:
+        return key
+    # "stop" is a special key that can be passed as input but is not used to
+    # format the prompt.
+    prompt_input_keys = list(set(inputs).difference(["stop"]))
+    if len(prompt_input_keys) != 1:
+        raise ValueError(f"One input key expected got {prompt_input_keys}")
+    return prompt_input_keys[0]
+
+
+class SimpleChatMemory(Memory, ChatMemory):
+    input_key: Optional[str] = None
+    output_key: Optional[str] = None
+
+    def clear(self) -> None:
+        self.clear()
+
+    @property
+    def memory_variables(self) -> List[str]:
+        return []
+
+    def load_memory_variables(self, inputs: Dict[str, Any]) -> Dict[str, str]:
+        return {}
```

Comment:
What's the idea behind this, ie. when would this be useful?

Reply:
useful for loading in memory at the start of a chain